### PR TITLE
More accessibility fixes

### PR DESCRIPTION
**Pull request by mullermp**
* Adds support for 320 px viewing (except tables)
* Support keyboard navigation of full_list
* Adds aria labels
* Bug fix for searching (introduced when upgrading jquery locally)
* Reads out search results

### CODE REVIEW

File: doc-src/templates/default/fulldoc/html/js/full_list.js [hunk 1]
```diff
@@ -0,0 +1,246 @@
+(function() {
```
Comment (mullermp) [inline doc-src/templates/default/fulldoc/html/js/full_list.js:3]: This file comes from https://github.com/lsegal/yard/blob/main/templates/default/fulldoc/html/js/full_list.js but has some modifications.

Here is the diff:

```
64a65,68
>     // custom - flip aria attribute
>     $(this).attr('aria-expanded', function (i, attr) {
>         return attr == 'true' ? 'false' : 'true'
>     });
66a71,84
> 
>   // custom - needed for accessibility
>   $('#full_list a.toggle').on('keypress',function(evt) {
>     // enter key
>     if (evt.which == 13) {
>       evt.stopPropagation();
>       evt.preventDefault();
>       $(this).parent().parent().toggleClass('collapsed');
>       $(this).attr('aria-expanded', function (i, attr) {
>           return attr == 'true' ? 'false' : 'true'
>       });
>       highlight();
>     }
>   });
94c112,113
<   $('#full_list').after("<div id='noresults' style='display:none'></div>");
---
>   // custom - add role status
>   $('#full_list').after("<div id='noresults' role='status' style='display:none'></div>");
144a164
> 
157c177,179
<   if ($('#full_list li:visible').size() === 0) {
---
>   // bug fix
>   var found = $('#full_list li > div.found').length;
>   if (found === 0) {
160c182,183
<     $('#noresults').text('').hide();
---
>     // accessibility - read out results
>     $('#noresults').text('There are ' + found + ' results.').hide().fadeIn();
190a214,220
> 
>   // custom - flip aria attribute
>   $target.find('a.toggle').attr('aria-expanded', 'true')
>   $target.parentsUntil('#full_list', 'li').each(function(i, el) {
>     $(el).find('> div > a.toggle').attr('aria-expanded', 'true');
>   });
> 
216c246
< })();
---
> })();
\ No newline at end of file
```

File: .yardopts [hunk 2]
```diff
@@ -6,4 +6,5 @@
 --hide-api private
 --plugin sitemap
 --exclude \/client_api.rb$
-gems/**/lib/**/*.rb
+gems/aws-sdk-core/lib/**/*.rb
+gems/aws-sdk-s3/lib/**/*.rb
```
Comment (alextwoods) [inline .yardopts]: temporary?

File: doc-src/templates/default/fulldoc/html/full_list.erb [hunk 3]
```diff
@@ -28,7 +28,7 @@
         </div>
 
         <!-- Add label for search -->
-        <div id="search">
+        <div id="search" >
```
Comment (alextwoods) [inline doc-src/templates/default/fulldoc/html/full_list.erb]: Is the space significant?

File: doc-src/templates/default/fulldoc/html/js/full_list.js (hunk 1 above)
Comment (alextwoods) [reply doc-src/templates/default/fulldoc/html/js/full_list.js:3]: Maybe add a comment at the top of the file about the source.

**Review by alextwoods — APPROVED** [review]
LGTM!

File: .yardopts (hunk 2 above)
Comment (mullermp) [reply .yardopts]: Yeah, for you to test.

File: doc-src/templates/default/fulldoc/html/full_list.erb (hunk 3 above)
Comment (mullermp) [reply doc-src/templates/default/fulldoc/html/full_list.erb]: No. I was doing something here before. I've removed it.

File: doc-src/templates/default/fulldoc/html/js/full_list.js (hunk 1 above)
Comment (mullermp) [reply doc-src/templates/default/fulldoc/html/js/full_list.js:3]: Done

**Pull request merged**